apps/gp2-server/test/data-providers/project.data-provider.test.ts

```diff
@@ -112,83 +113,129 @@ describe('Project Data Provider', () => {
-        const project = getGraphQLProject();
-        project.flatData.members = [getGraphQLProjectMember()];
-        const { members } = parseProjectToDataObject(project);
-        expect(members).toEqual([
-          {
-            userId: '42',
-            firstName: 'Tony',
-            lastName: 'Stark',
-          },
-        ]);
-      });
-
-      test('undefined members returns empty array', () => {
-        const project = getGraphQLProject();
-        project.flatData.members = undefined!;
-        const { members } = parseProjectToDataObject(project);
-        expect(members).toEqual([]);
-      });
-
-      test('avatar urls are added if available', () => {
-        const project = getGraphQLProject();
-        const member = getGraphQLProjectMember();
-        member!.user![0]!.flatData.avatar = [{ id: 'avatar-id' }];
-        project.flatData.members = [member];
-        const { members } = parseProjectToDataObject(project);
-        expect(members[0]?.avatarUrl).toEqual(
-          `${baseUrl}/api/assets/${appName}/avatar-id`,
-        );
-      });
-
-      test('should skip the user from the result if the user property is undefined', () => {
-        const project = getGraphQLProject();
-        const member = getGraphQLProjectMember();
-        member!.user = undefined!;
-        project.flatData.members = [member];
-        const { members } = parseProjectToDataObject(project);
-        expect(members).toEqual([]);
-      });
-
-      test('pm emails are added if available', () => {
-        const email = 'tony@starkenterprises.com';
-        const project = getGraphQLProject();
-        project.flatData.pmEmail = email;
-        const { pmEmail } = parseProjectToDataObject(project);
-        expect(pmEmail).toEqual(email);
-      });
-
-      test('lead emails are added if available', () => {
-        const email = 'tony@starkenterprises.com';
-        const project = getGraphQLProject();
-        project.flatData.leadEmail = email;
-        const { leadEmail } = parseProjectToDataObject(project);
-        expect(leadEmail).toEqual(email);
-      });
-
-      test('description is added if available', () => {
-        const expectedDescription = 'this is a description';
-        const project = getGraphQLProject();
-        project.flatData.description = expectedDescription;
-        const { description } = parseProjectToDataObject(project);
-        expect(description).toEqual(expectedDescription);
-      });
-
-      test('keywords are added', () => {
-        const expectedKeywords = ['Outreach'];
-        const project = getGraphQLProject();
-        project.flatData.keywords = expectedKeywords;
-        const { keywords } = parseProjectToDataObject(project);
-        expect(keywords).toEqual(expectedKeywords);
+      describe('members', () => {
+        test('the members in project are parsed', () => {
+          const Project = getGraphQLProject();
+          Project.flatData.members = [getGraphQLProjectMember()];
```

Comment:
maybe the project fixture should contain a member by default?

Reply:
So in other place we have been leaving optional properties empty/undefined. i.e. https://github.com/yldio/asap-hub/blob/a87ef8a2ee726f144baba1d97432b795ea3202ef/apps/gp2-server/test/fixtures/working-group.fixtures.ts#L19

We then explicitly set them in test. One to discuss I think.